petals-esb-deployer-impl/src/main/java/org/ow2/petals/deployer/runtimemodel/RuntimeContainer.java

```diff
@@ -140,8 +149,46 @@ public RuntimeComponent getComponent(final String id) {
         return components.get(id);
     }
 
+    /**
+     * Adding or removing from the returned collection does not affect component.
+     */
     public Collection<RuntimeComponent> getComponents() {
-        return components.values();
+        return new HashSet<>(components.values());
```

Comment:
I think you can find better using class Collections.unmodifiableCollection() to avoid adding and removing from the returned collection.